### DISCUSSION TITLE
Switched accounts in Github Desktop but commits are showing up as old account

### DISCUSSION THREAD

**Discussion by whscienceteacher** [community/community#22250] [New to GitHub]
<p>I logged out through github desktop from Account A and in with Account B. However I just noticed that commits to my repository owned by Account B are shown as from Account A.</p>
<p>How can I fix this so they show from Account B? Is there anyway to retroactively change the owner of the commits?</p>
<p>(I’m new to this so I am not sure I am using the terms correctly)</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>You’ll have to update your author email address, which Github uses to assign commits to an account. You might want to also update your author name to match. Note that you can set these globally, but you can also override the global value for a particular repository, which might be useful if you use different identities for different repositories.<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address#setting-your-commit-email-address-in-git" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="16" height="16">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address#setting-your-commit-email-address-in-git" target="_blank" rel="noopener">Setting your commit email address - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>You can change past commits by rewriting history. Note that this will make things complicated for anyone you have shared the repository with, they will have to manually accept the change. Depending on how many commits you want to change you might either used <code>git rebase -i</code> and edit each commit in turn, or use a bulk approach with <code>git filter-branch</code> (described at the end of the article linked below).<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://git-scm.com/favicon.ico" class="site-icon" width="16" height="16">
      <a href="https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History" target="_blank" rel="noopener">git-scm.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="16" height="16">

<h3><a href="https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History" target="_blank" rel="noopener">Git - Rewriting History</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
